zerver/worker/thumbnail.py

```diff
@@ -120,9 +127,46 @@ def ensure_thumbnails(image_attachment: ImageAttachment) -> int:
             # We have never thumbnailed this -- it most likely had
             # bad data.  Remove the ImageAttachment row, since it is
             # not valid for thumbnailing.
+            update_message_rendered_content(
+                image_attachment.realm_id, image_attachment.path_id, None
+            )
             image_attachment.delete()
+            # TODO: Clean up any dangling thumbnails we may have
+            # produced?  Seems unlikely that we'd fail on one size,
+            # but not another, but anything's possible.
```

Comment:
Hmm, if `written_images` is 0, I think that may mean we can't have leaked any? But maybe we need cleanup logic for an `else` clause on that detail?

Reply:
Ah, yeah, this would need to be an adjacent `else`.